libpod/container.go

```diff
@@ -536,12 +536,63 @@ func (c *Container) Unmount() error {
 
 // Pause pauses a container
 func (c *Container) Pause() error {
-	return ErrNotImplemented
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if err := c.syncContainer(); err != nil {
+		return err
+	}
+
+	if c.state.State == ContainerStatePaused {
+		return errors.Wrapf(ErrCtrStateInvalid, "%q is already paused", c.ID())
+	}
+	if c.state.State == ContainerStateRunning {
+		if err := c.runtime.ociRuntime.pauseContainer(c); err != nil {
+			return err
+		}
+
+		logrus.Debugf("Paused container %s", c.ID())
+
+		// Update container's state as it should be ContainerStatePaused now
+		if err := c.runtime.ociRuntime.updateContainerStatus(c); err != nil {
+			return err
+		}
+
+		if err := c.runtime.state.SaveContainer(c); err != nil {
+			return errors.Wrapf(err, "error saving container %s state", c.ID())
+		}
+		return nil
+	}
+	return errors.Wrapf(ErrCtrStateInvalid, "%q is not running, can't pause", c.ID())
 }
 
 // Unpause unpauses a container
 func (c *Container) Unpause() error {
-	return ErrNotImplemented
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	if err := c.syncContainer(); err != nil {
+		return err
+	}
+
+	if c.state.State == ContainerStatePaused {
```

Comment:
Instead of wrapping all the code in an if(), can you have a single if after syncContainer() to error out if state != ContainerStatePaused

Reply:
fixed